distributed/utils.py

```diff
@@ -371,17 +371,27 @@ def in_async_call(loop, default=False):
         return False
 
 
-def sync(loop, func, *args, callback_timeout=None, **kwargs):
+def sync(
+    loop: IOLoop,
+    func: Callable[..., Awaitable[T]],
+    *args: AnyType,
+    callback_timeout: str | float | timedelta | None = None,
+    **kwargs: AnyType,
```

Comment:
```suggestion
    func: Callable[P, Awaitable[T]],
    *args: P.args,
    callback_timeout: str | float | timedelta | None = None,
    **kwargs: P.kwargs,
```

add at the top of the module:
```python
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    # TODO import from typing (requires Python >=3.10)
    from typing_extensions import ParamSpec

    P = ParamSpec("P")
```

Reply:
From what I understand, `ParamSpec` doesn't work in combination with additional keyword arguments (`callback_timeout`). At least, that's what I gathered from the PEP and trying to get this to work. LMK if you know how to pull this off!

Reply:
You can get this working with just the *args with TypeVarTuple and Unpack but you'll need to opt into that support on mypy 1.6

This TypeVarTuple and Unpack will show up as Any to users without this support so it's safe to opt in here

Reply:
Let's upgrade mypy and utilize `TypeVarTuple` in a follow-up PR.